src/odin/maneuversbuilder.cc

```diff
@@ -3255,7 +3257,9 @@ void ManeuversBuilder::ProcessTurnLanes(std::list<Maneuver>& maneuvers) {
     // on to the next maneuver...
     prev_man = curr_man;
     curr_man = next_man;
-    ++next_man;
+    if (next_man != maneuvers.end()) {
```

Comment:
loop condition is `while (curr_man != maneuvers.end())` here, i.e. there is a case when `next_man` points to end of the list.

Reply:
i think it would help to explain what is really going on here. i dont think its undefined behavior to increment an iterator past the end but it is undefined behavior to deference it. we should note here that there is a some code here that checks `next_man` before using it, but it checks for equality `if (next_man != maneuvers.end())`. of course that check will fail when we iterate past the end and we'll dereference it and be screwed. maybe we should just change the other code to: `if (next_man < maneuvers.end())`?

Reply:
Yup, this fix looks correct since the loop is only checking validity of `curr_man` and not `next_man`.